plugins/inputs/jboss/jboss.go

```diff
@@ -0,0 +1,1312 @@
+
+	client HTTPClient
+}
+
+// HTTPClient HTTP client struct
+type HTTPClient interface {
+	// Returns the result of an http request
+	//
+	// Parameters:
+	// req: HTTP request object
+	//
+	// Returns:
+	// http.Response:  HTTP respons object
+	// error        :  Any error that may have occurred
+	MakeRequest(req *http.Request) (*http.Response, error)
+
+	SetHTTPClient(client *http.Client)
+	HTTPClient() *http.Client
+}
+
+// RealHTTPClient the HTTP client handler
+type RealHTTPClient struct {
+	client *http.Client
+}
+
+// MakeRequest do an HTTP request
+func (c *RealHTTPClient) MakeRequest(req *http.Request) (*http.Response, error) {
+	return c.client.Do(req)
+}
+
+// SetHTTPClient set http client
+func (c *RealHTTPClient) SetHTTPClient(client *http.Client) {
+	c.client = client
+}
+
+// HTTPClient return the HTTP client
+func (c *RealHTTPClient) HTTPClient() *http.Client {
+	return c.client
+}
+
+var sampleConfig = `
+  # Config for get statistics from JBoss AS
+  servers = [
+    "http://localhost:9090/management",
+  ]
+  
+  ## Username and password
+  username = ""
+  password = ""
+  
+  ## authorization mode could be "basic" or "digest"
+  authorization = "digest"
+
+  ## Optional SSL Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use SSL but skip chain & host verification
+  # insecure_skip_verify = false
+  ## Metric selection
+  metrics =[
+	"jvm",
+	"web", 		# Handles both EAP <=6.X/AS <=7.X and EAP >=7.X/Widlfly > 8
+	"deployment",
+	"database",
+	"transaction",
+	"jms",
+  ]
+`
+
+// SampleConfig returns a sample configuration block
+func (*JBoss) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description just returns a short description of the JBoss plugin
+func (*JBoss) Description() string {
+	return "Telegraf plugin for gathering metrics from JBoss AS"
+}
+
+// Gather Gathers data for all servers.
+func (h *JBoss) Gather(acc telegraf.Accumulator) error {
+
+	if h.ResponseTimeout.Duration < time.Second {
+		h.ResponseTimeout.Duration = time.Second * 5
+	}
+
+	if h.client.HTTPClient() == nil {
+		tlsCfg, err := h.ClientConfig.TLSConfig()
+
+		if err != nil {
+			return err
+		}
+		tr := &http.Transport{
+			TLSClientConfig: tlsCfg,
+		}
+		client := &http.Client{
+			Transport: tr,
+			Timeout:   h.ResponseTimeout.Duration,
+		}
+		h.client.SetHTTPClient(client)
+	}
+
+	var wg sync.WaitGroup
+	for _, server := range h.Servers {
+
+		//Check Exec Mode for each servers
+
+		bodyContent, err := h.createRequestBody(GetExecStat, nil)
+		if err != nil {
+			acc.AddError(err)
+		}
+
+		out, err := h.doRequest(server, bodyContent)
+		if err != nil {
+			log.Printf("E! JBoss Error handling ExecMode Test: %s\n", err)
+			acc.AddError(err)
+		}
+		// Unmarshal json
+		exec := ExecTypeResponse{}
+		if err = json.Unmarshal(out, &exec); err != nil {
+			acc.AddError(fmt.Errorf("Error decoding JSON response (ExecTypeResponse) %s,%s", out, err))
```

Comment:
Errors should start lowercase. Maybe
```suggestion
			acc.AddError(fmt.Errorf("decoding JSON response (ExecTypeResponse) %q failed: %v", out, err))
```

Reply:
Still valid.